buildSrc/src/main/java/org/elasticsearch/gradle/test/RestIntegTestTask.java

```diff
@@ -0,0 +1,114 @@
+import org.elasticsearch.gradle.testclusters.ElasticsearchCluster;
+import org.elasticsearch.gradle.testclusters.RestTestRunnerTask;
+import org.gradle.api.Action;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.NamedDomainObjectContainer;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+
+import java.util.function.Supplier;
+
+public class RestIntegTestTask extends DefaultTask {
+
+    protected RestTestRunnerTask runner;
+    private static final String TESTS_REST_CLUSTER = "tests.rest.cluster";
+    private static final String TESTS_CLUSTER = "tests.cluster";
+    private static final String TESTS_CLUSTER_NAME = "tests.clustername";
+
+    public RestIntegTestTask() {
+        Project project = getProject();
+        String name = getName();
+        super.dependsOn(project);
+        runner = project.getTasks().create(name + "Runner", RestTestRunnerTask.class);
+        NamedDomainObjectContainer<ElasticsearchCluster> testClusters = (NamedDomainObjectContainer<ElasticsearchCluster>) project
+            .getExtensions()
+            .getByName("testClusters");
+        ElasticsearchCluster cluster = testClusters.create(name);
+        runner.useCluster(cluster);
+        runner.include("**/*IT.class");
+        runner.systemProperty("tests.rest.load_packaged", Boolean.FALSE.toString());
+        if (System.getProperty(TESTS_REST_CLUSTER) == null) {
+            if (System.getProperty(TESTS_CLUSTER) != null || System.getProperty(TESTS_CLUSTER_NAME) != null) {
+                throw new IllegalArgumentException(
+                    String.format("%s, %s, and %s must all be null or non-null", TESTS_REST_CLUSTER, TESTS_CLUSTER, TESTS_CLUSTER_NAME)
+                );
+            }
+            SystemPropertyCommandLineArgumentProvider runnerNonInputProperties = (SystemPropertyCommandLineArgumentProvider) runner
+                .getExtensions()
+                .getByName("nonInputProperties");
+            runnerNonInputProperties.systemProperty(
+                TESTS_REST_CLUSTER,
+                (Supplier<String>) () -> String.join(",", cluster.getAllHttpSocketURI())
+            );
+            runnerNonInputProperties.systemProperty(
+                TESTS_CLUSTER,
+                (Supplier<String>) () -> String.join(",", cluster.getAllTransportPortURI())
+            );
+            runnerNonInputProperties.systemProperty(TESTS_CLUSTER_NAME, (Supplier<String>) cluster::getName);
+        } else {
+            if (System.getProperty(TESTS_CLUSTER) == null || System.getProperty(TESTS_CLUSTER_NAME) == null) {
+                throw new IllegalArgumentException(
+                    String.format("%s, %s, and %s must all be null or non-null", TESTS_REST_CLUSTER, TESTS_CLUSTER, TESTS_CLUSTER_NAME)
+                );
+            }
+            // an external cluster was specified and all responsibility for cluster configuration is taken by the user
```

Comment:
I'm now not sure this is even necessary. We already by default pass thru any system properties with the prefix `tests`.

https://github.com/elastic/elasticsearch/blob/ecdbd37f2e0f0447ed574b306adb64c19adc3ce1/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy#L676-L680

Reply:
I am pretty sure you are right. The fact that it also had a bug that went un-noticed means that either no one used it, or it did in fact work. (I swear I had used this before without issue). Also if assume code hasn't moved too much, it looks like the referenced passthrough happened after the introduction here. 

I have removed it, and will do a quick manual spot check to ensure it still works. (I assume it does and will only post otherwise)